### PR TITLE
Fix OOI edit form

### LINKED ISSUE #1495: AttributeError when editing an IPv4Address object
**Describe the bug**
When attempting to edit an IPv4 address and saving it, the following error occurs:

image

**To Reproduce**
1. Make sure an IPV4Address object exists
2. Edit it through the object detail view
3. Press save

**Expected behavior**
No changes, no error.

### PR DESCRIPTION
**Pull request by praseodym**
### Changes

**Convert to string in OOI edit form**
    
Currently, the form can have e.g. an `ipaddress.IPv4Address` passed in,
causing errors when saving the form.
    
There are some exceptions to this, such as dicts used for JSON fields
(Config OOIs).

**Fix initial value for enum fields**

**Clean up OOI edit form template**

### Issue link
Fixes #1495

### Proof
n/a

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/rocky/views/ooi_edit.py [hunk 1]
```diff
@@ -17,12 +16,10 @@ def get_initial(self):
         initial = super().get_initial()
 
         for attr, value in self.ooi:
-            if isinstance(value, Reference):
-                initial[attr] = str(value)
-            elif isinstance(value, list):
+            if isinstance(value, list):
```
[the remark this answers is not in the document]
Comment (praseodym) [reply rocky/rocky/views/ooi_edit.py:19]: I don't know where we pass in lists, let alone sets or tuples. I can add it but I'm not sure if we would use it?

**Comment by praseodym** [comment]
> I'm afraid that the JSON Fields will not work anymore with this fix. Could you verify?

Fixed this by excluding `dict` values from being converted to strings.

**Comment by noamblitz** [comment]
> > I'm afraid that the JSON Fields will not work anymore with this fix. Could you verify?
> 
> Fixed this by excluding `dict` values from being converted to strings.

And ints? Maybe we should make the same exceptions as in the add ooi form

**Comment by praseodym** [comment]
> And ints? Maybe we should make the same exceptions as in the add ooi form

What exceptions do you mean?

**Comment by praseodym** [comment]
Editing an IPPort OOI also fails on both `main` and this branch, with error "Error: value is not a valid enumeration member; permitted: 'tcp', 'udp'". Let's take a closer look to how these forms are supposed to work next week.

**Comment by underdarknl** [comment]
> Editing an IPPort OOI also fails on both `main` and this branch, with error "Error: value is not a valid enumeration member; permitted: 'tcp', 'udp'". Let's take a closer look to how these forms are supposed to work next week.

Maybe we could produce a jsonschema for the models currently used and render that / check the returned data against that schema? This would also be helpful in moving to models defined as json anyway.

**Comment by praseodym** [comment]
> Editing an IPPort OOI also fails on both `main` and this branch, with error "Error: value is not a valid enumeration member; permitted: 'tcp', 'udp'". Let's take a closer look to how these forms are supposed to work next week.

This is fixed by handling the initial value for `Enum` types.

> Maybe we could produce a jsonschema for the models currently used and render that / check the returned data against that schema? This would also be helpful in moving to models defined as json anyway.

I think redoing the entire OOI model architecture would be a lot of work without a lot of benefits on the short term.

File: rocky/rocky/views/ooi_edit.py [hunk 2]
```diff
@@ -17,12 +18,15 @@ def get_initial(self):
...
                 initial[attr] = [str(x) for x in value]
-            else:
+            if isinstance(value, Enum):
```
Comment (ammar92) [inline rocky/rocky/views/ooi_edit.py:23]:
```suggestion
            elif isinstance(value, Enum):
```

**Comment by Darwinkel** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* Editing and saving IPAddressV4 and IPPort
* No apparent regressions

**Review by underdarknl — APPROVED** [review]

**Pull request merged**